garden_ai/utils.py

```diff
@@ -196,3 +201,105 @@ def f_of_g(*args, **kwargs):
     )
     f_of_g.__name__ = f.__name__ + "_COMPOSED_WITH_" + g.__name__
     return f_of_g
+
+
+def mint_doi(
+    title: str,
+    authors: List[str],
+    contributors: List[str],
+    year: str,
+    description: str = "",
+    publisher: str = "thegardens.ai",
+    resourceType: str = "AI/ML",
+    test: bool = True,
+):
+    """Request a "findable" DOI from DataCite
+
+    Note that this expects environment variables 'DATACITE_REPOSITORY_ID', 'DATACITE_PASSWORD' and
+    'DOI_PREFIX' to be set correctly in order to authenticate with DataCite.
+
+    Parameters
+    ----------
+    title : str
+        Title that should appear in citation
+    authors : list[str]
+    contributors: list[str]
+    year : str
+        YYYY
+    description: str
+        free-text description of the Digital Object being made Identifiable
+    publisher : str
+        "Garden" seems like a sane default?
+    resourceType : str
+        "AI/ML" by default -- this is allowed to be free text by DataCite but is advised to be kept short
+    test : bool
+        Whether to hit the DataCite 'test' API or the real one - currently only the former should be
+        used as it's the only one implemented.
+
+    Raises
+    ------
+    HTTPError
+        if one is encountered by `requests.Response.raise_for_status`
+
+    NotImplementedError
+        if given `test=False`
+    """
+    if not test:
```

Comment:
To the extent possible, it's nice to avoid having application code be aware that it's under test. I would make the doi prefix and datacite urls environment variables. There could be local default settings where they are the test values. A prod deployment would have them set to the real values.